app/controllers/candidates_controller.rb

```diff
@@ -1,33 +1,31 @@
 class CandidatesController < ApplicationController
   def show
+    @position = Position.find(params[:position_id])
     @candidate = Candidate.find(params[:id])
```

Comment:
better to make callback to set hiring position and find candidates against a hiring position instead of Candidate.find

Reply:
done, replaced with @candidate = @position.candidates.find(params[:id]) everywhere
and finding position in before action callback